pandas/indexes/multi.py

```diff
@@ -71,8 +71,10 @@ def __new__(cls, levels=None, labels=None, sortorder=None, names=None,
                 name=None, **kwargs):
 
         # compat with Index
-        if name is not None:
-            names = name
+        if name is None and hasattr(levels, 'name'):
+            name = levels.name
+        if isinstance(levels, MultiIndex):
+            return levels.copy(name=name, names=names, deep=copy)
```

Comment:
It is tested [here](https://github.com/pydata/pandas/pull/13205/files#diff-9ae0a94014431c93e23da48433811c49R743)